lib/rarity-map.js

```diff
@@ -1,121 +1,108 @@
-module.exports = RarityMap
 
 /**
  * Mapping of torrent pieces to their respective availability in the torrent swarm. Used
  * by the torrent manager for implementing the rarest piece first selection strategy.
  */
-function RarityMap (torrent) {
-  var self = this
-
-  self._torrent = torrent
-  self._numPieces = torrent.pieces.length
-  self._pieces = []
+class RarityMap {
+  constructor (torrent) {
+    this._torrent = torrent
+    this._numPieces = torrent.pieces.length
+    this._pieces = new Array(this._numPieces).fill(0)
```

Comment:
What is the fill here for? If it's not needed I think it will just consume O(n) to fill it and can be removed.